pymongo/synchronous/srv_resolver.py

```diff
@@ -137,7 +137,8 @@ def _get_srv_response_and_hosts(
 
         # Validate hosts
         for node in nodes:
-            if self.__fqdn == node[0].lower():
+            srv_host = node[0].lower()
+            if self.__fqdn == srv_host and len(srv_host.split(".")) < 3:
```

Comment:
Can we use `self.__slen` instead of `srv_host.split(".")`? We could also remain `__slen` for clarify.

Reply:
And I don't know why this class is used `__` attributes, this is a private class anyway so we can rename `self.__fqdn` -> `self.fqdn` and `self.__slen` -> `self.nparts` or something.

But up to you, I'm less concerned about the naming in this PR. My bigger point is that the check here should be based on the length of FQDN, not the length of the response, to match the spec's description.